### DISCUSSION TITLE
How can I describe entity relation unidirectional one-to-many?

### DISCUSSION THREAD

**Discussion by jays1204** [mikro-orm/mikro-orm#2928] [Q&A] [answered]
In Mikro-ORM v5, How can I describe entity relation unidirectional one-to-many?

for example, User has N-Posts. User can access Posts, but I don't want to access User from Post.
But I can not find example or solution for one-to-many relation with unidirectional. Because I want to avoid bidirectional relation.
Could you give me examples and documents to solve this problem?

```
@Entity()
export class User {
  @PrimaryKey()
  userId: number;

  @OneToMany() // <- How do I set it up?
  posts = new Collection<Post>(this);

}

@Entity()
export class Post {
  @PrimaryKey()
  postId: number;

  @Property()
  contents!: string;
}
```

**Comment by B4nan** [comment, marked as answer]
There are no examples, because it is not possible. 1:m is a virtual collection based on the m:1 relation.

**Reply by mujehoxe** [reply]
It should be possible, the only requirement is to provide the foreign key without the need for the many entity to have a reference to the one entity.
Here is an example from hibernate
https://stackoverflow.com/questions/12038380/how-to-define-unidirectional-onetomany-relationship-in-jpa

**Reply by B4nan** [reply]
> It should be possible, the only requirement is to provide the foreign key without the need for the many entity to have a reference to the one entity.

You need a FK, that's how it works here - the OP was asking for a FK-less solution, which does not make sense to me. You can use `mapToPk` if you want to work with the PK value rather than mapping it to entity instance.

https://mikro-orm.io/docs/5.4/defining-entities#mapping-directly-to-primary-keys

**Reply by xenoterracide** [reply]
does a mongo aggregate need an fk? even if you need an fk, why do you need a property on the other side... why not just be able to specify the db level fk as a property.
